ballerina/tests/enum_type_test.bal

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+
+enum Color {
+    RED,
+    GREEN,
+    BLUE
+}
+
+const OPEN = "open";
+const CLOSE = "close";
+type STATE OPEN|CLOSE;
```

Comment:
```suggestion
type State OPEN|CLOSE;
```

How does something like this work?

```ballerina
type State OPEN|CLOSE;

type OptionalState OPEN|CLOSE?
```

We should consider this for the other way of defining enums as well.

Reply:
Yes added support for this as well, will update with a test case.

Reply:
Added.